sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt

```diff
@@ -162,101 +177,112 @@ data class ProfileVendorAdapter(
+                        .left()
+            }
 
-                                when {
-                                    simEntryId == null -> AdapterError("simEntryId == null").left()
-                                    statusEid == null -> AdapterError("statusEid == null").left()
-                                    else -> dao.setEidOfSimProfile(simEntryId, statusEid)
-                                }
-                            }
-                            if (!eid.isNullOrEmpty() && eid != status.eid) {
-                                logger.warn("EID returned from service {} does not match provided EID ({} <> {}) in SM-DP+ 'order-confirm' message (call-id: {})",
-                                        config.name,
-                                        eid,
-                                        status.eid,
-                                        header.functionCallIdentifier)
-                            }
-                            logger.info("SM-DP+ 'order-confirm' message to service {} for ICCID {} completed OK (call-id: {})",
-                                    config.name,
-                                    simEntry.iccid,
-                                    header.functionCallIdentifier)
+            if (status.eid.isNullOrEmpty()) {
+                logger.warn("No EID returned from service {} for ICCID {} for SM-DP+ 'order-confirm' message (call-id: {})",
+                        nameOfSimVendor,
+                        iccid,
+                        functionCallIdentifier)
+            } else {
 
-                            val simEntryId = simEntry.id
-                            val statusMatchingId = status.matchingId
+                val simEntryId = simEntry.id
+                val statusEid = status.eid
 
-                            when {
-                                simEntryId == null -> AdapterError("simEntryId == null").left()
-                                statusMatchingId == null -> AdapterError("statusMatchingId == null").left()
-                                else -> dao.setSmDpPlusStateAndMatchingId(simEntryId, SmDpPlusState.RELEASED, statusMatchingId)
-                            }
-                        }
-                    }
-                    else -> {
-                        logger.error("SM-DP+ 'order-confirm' message to service {} for ICCID {} failed with status code {} (call-id: {})",
-                                config.name,
-                                simEntry.iccid,
-                                it?.statusLine?.statusCode,
-                                header.functionCallIdentifier)
-                        NotUpdatedError("SM-DP+ 'order-confirm' to ${config.name} failed with code: ${it.statusLine.statusCode}")
-                                .left()
-                    }
+                when {
+                    simEntryId == null -> AdapterError("simEntryId == null").left()
```

Comment:
Not sure about this. The `simEntryId` and `statusId` goes out of scope after the `else` part...

Reply:
Have simplified the code.